crates/api_common/src/utils.rs

```diff
@@ -141,18 +145,20 @@ pub async fn mark_post_as_unread(
     .map_err(|e| LemmyError::from_error_message(e, "couldnt_mark_post_as_read"))
 }
 
-// TODO: this should simply take LemmyContext as param
 #[tracing::instrument(skip_all)]
-pub async fn get_local_user_view_from_jwt(
-  jwt: &str,
-  pool: &DbPool,
-  secret: &Secret,
+pub async fn local_user_view_from_jwt(
+  auth: Option<Sensitive<String>>,
+  context: &LemmyContext,
 ) -> Result<LocalUserView, LemmyError> {
-  let claims = Claims::decode(jwt, &secret.jwt_secret)
-    .map_err(|e| e.with_message("not_logged_in"))?
+  let auth = auth.ok_or(LemmyError::unauthorized())?;
```

Comment:
If we accidentally use this version for one of the really optional auth routes, its going to fail. 

Essentially you've moved the auth checking from required for some routes, to optional for them all, and now any route could run into this error if we don't correctly use the optional version in code. Definitely scary and a little bit more difficult to maintain, since this removes a compile-time check.